lib/puppet/util.rb

```diff
@@ -758,6 +758,17 @@ def deterministic_rand_int(seed,max)
     Random.new(seed).rand(max)
   end
   module_function :deterministic_rand_int
+
+  # Executes a block of code, wrapped around Facter.load_external(false) and Facter.load_external(true) which will cause Facter to not evaluate external facts.
+  def skip_external_facts
+    begin
+      Facter.load_external(false) if Facter.respond_to? :load_external
+      yield
+    rescue
+      Facter.load_external(true) if Facter.respond_to? :load_external
```

Comment:
you can return early, instead of checking each time
```ruby
return unless Facter.respond_to? :load_external
```
then execute the rest of the code